### PR TITLE
Add unittests for crawl processing

### PR DESCRIPTION
**Pull request by sammacbeth**
Simple tests for processing a crawl. The intention is to act as documentation of how crawl data is processed into tracker-radar stats.

### CODE REVIEW

**Comment by sammacbeth** [comment]
These tests currently fail on CI, because [sharedData](https://github.com/duckduckgo/tracker-radar-detector/blob/main/src/trackers/helpers/sharedData.js#L18) does not load without a config that points to a tracker-radar repo. As this file is imported by multiple classes, we cannot import them for tests without this breaking.

A solution to this would be to pass sharedData in to these classes at runtime, however this requires quite a bit of refactoring of the code, which may be out of the scope of this PR. However, long term I'm not sure just catching errors loading the files and continue will be a good solution, as this will likely lead to downstream problems.

File: test/build-tracker.test.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    fpStd: 0,
+    host: "tracker.com",
+    prevalence: 1,
+    responseHashes: [],
+    rule: "tracker\\.com\\/collect",
+    sites: 2,
+    subdomains: new Set(['dummy']),
+    type: "XHR",
+}
+const commonRequestData2 = {
```
Comment (jdorweiler) [inline test/build-tracker.test.js:22]: Do these need `nameservers` since I just added those?

Comment (sammacbeth) [reply test/build-tracker.test.js:22]: This test doesn't currently test all attributes, but we could add tests for nameservers fairly easily here. However, to properly test the fetching of nameserver data we'd have to mock the ns lookups, as relying on real lookups in tests is not a great idea.

Comment (jdorweiler) [reply test/build-tracker.test.js:22]: Makes sense, thanks.

**Comment by jdorweiler** [comment]
👍

**Pull request merged**